tests/python/test_parser.py

```diff
@@ -933,3 +933,331 @@ def test_string_with_invalid_utf8_sequences(self):
 
     def test_parser_error(self):
         self.assertRaises(yaramod.ParserError, yaramod.parse_string, 'rule {')
+
+    def test_simple_regular_expression_in_strings(self):
+        yara_file = yaramod.parse_string(r'''
+rule rule_with_regexp {
+    strings:
+        $1 = /asd/
+    condition:
+        true
+}''')
+
+        self.assertEqual(len(yara_file.rules), 1)
+
+        rule = yara_file.rules[0]
+        self.assertEqual(rule.name, 'rule_with_regexp')
+        self.assertEqual(rule.modifier, yaramod.RuleModifier.Empty)
+        self.assertEqual(len(rule.metas), 0)
+        self.assertEqual(len(rule.strings), 1)
+        self.assertEqual(len(rule.tags), 0)
+
+        regexp_string = rule.strings[0]
+        self.assertEqual(regexp_string.identifier, '$1')
+        self.assertTrue(regexp_string.is_regexp)
+        self.assertTrue(isinstance(regexp_string.unit, yaramod.RegexpConcat))
+        self.assertEqual(regexp_string.text, '/asd/')
+        self.assertEqual(regexp_string.pure_text, b'asd')
+        self.assertEqual(regexp_string.suffix_modifiers, '')
+        self.assertEqual(regexp_string.modifiers_text, '')
+        self.assertEqual(len(regexp_string.unit.units), 3)
+        self.assertEqual(regexp_string.unit.units[0].text, 'a')
+        self.assertEqual(regexp_string.unit.units[1].text, 's')
+        self.assertEqual(regexp_string.unit.units[2].text, 'd')
+
+    def test_complex_regular_expression_in_strings(self):
+        yara_file = yaramod.parse_string(r'''
+rule rule_with_complex_regexp {
+    strings:
+        $1 = /asd|1234/ nocase ascii wide fullword
+        $2 = /md5: [0-9a-zA-Z]{32}/
+    condition:
+        true
+}''')
+
+        self.assertEqual(len(yara_file.rules), 1)
+
+        rule = yara_file.rules[0]
+        self.assertEqual(rule.name, 'rule_with_complex_regexp')
+        self.assertEqual(rule.modifier, yaramod.RuleModifier.Empty)
+        self.assertEqual(len(rule.metas), 0)
+        self.assertEqual(len(rule.strings), 2)
+        self.assertEqual(len(rule.tags), 0)
+
+        regexp_string = rule.strings[0]
+        self.assertEqual(regexp_string.identifier, '$1')
+        self.assertTrue(regexp_string.is_regexp)
+        self.assertEqual(regexp_string.text, '/asd|1234/ ascii wide nocase fullword')
+        self.assertEqual(regexp_string.pure_text, b'asd|1234')
+        # Looks like the suffix modifiers are stored in multiple places
+        # and the only ones that are used are from String class.
+        # self.assertEqual(regexp_string.suffix_modifiers, '')
```

Comment:
For what I mean with suffix modifiers, open https://perldoc.perl.org/perlre.html and scroll down to modifiers overview. YARA only supports `i` and `s` modifiers.